### PR TITLE
Diagnose me dev env

### PR DESCRIPTION
**Pull request by SinaChavoshi**
Adding libraries for diagnose me dev env .

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2425)

### CODE REVIEW

File: sdk/python/kfp/cli/diagnose_me/dev_env_test.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    mock_executor_response().execute_command.assert_called_with(
+        ['pip3', 'list', '--format', 'json'])
+
+  @mock.patch.object(utility, 'ExecutorResponse', autospec=True)
+  def test_dev_env_configuration_human_readable(self, mock_executor_response):
+    """Tests dev_env command execution."""
+    dev_env.get_dev_env_configuration(
+        dev_env.Commands.PIP3LIST, human_readable=True)
+    mock_executor_response().execute_command.assert_called_with(
+        ['pip3', 'list'])
```
Comment (Ark-kun) [inline sdk/python/kfp/cli/diagnose_me/dev_env_test.py]: Should we also use `python3 -m pip` here?

P.S. Maybe we should report the paths of both (pip3 and python3) as part of diagnostics. Also the versions (`python3 --version`; `pip3 --version`; `python3 -m pip --version`)

File: sdk/python/kfp/cli/diagnose_me/dev_env.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+class Commands(enum.Enum):
+  """Enum for gcloud and gsutil commands."""
+  PIP3LIST = 1
+  PYTHON3PIPLIST = 2
+  PIP3VERSION = 3
+  PYHYON3PIPVERSION = 4
+
+
+_command_string = {
```
Comment (Ark-kun) [inline sdk/python/kfp/cli/diagnose_me/dev_env.py]: Maybe we should report the paths of both (pip3 and python3) as part of diagnostics.

Comment (SinaChavoshi) [reply sdk/python/kfp/cli/diagnose_me/dev_env.py]: Agreed :) updated accordingly.

**Pull request merged**